pilot/pkg/config/kube/crd/client.go

```diff
@@ -137,11 +138,15 @@ func (rc *restClient) createRESTConfig(kubeconfig string) (config *rest.Config,
 	if kubeconfig == "" {
 		config, err = rest.InClusterConfig()
 	} else {
-		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+		loadingRules.ExplicitPath = kubeconfig
+
+		clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
+		config, err = clientConfig.ClientConfig()
```

Comment:
Looks like [`BuildConfigFromFlags`](https://github.com/kubernetes/client-go/blob/b4efb2d46b33411d25a37b364098993ca896e900/tools/clientcmd/client_config.go#L545) handles all of these cases (e.g. in and out of cluster plus default client loading rules). Using that in istioctl might also fix https://github.com/istio/istio/issues/1971.